src/containers/NewNote/NewNote.tsx

```diff
@@ -1,23 +1,53 @@
-import React, { useRef, useState, FormEvent, ChangeEvent, FC } from 'react';
+import React, {
+  useRef,
+  useState,
+  FormEvent,
+  ChangeEvent,
+  FC,
+  useEffect
+} from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import { RouteComponentProps } from 'react-router';
 import { Button, Container, TextField } from '@material-ui/core';
-import { API } from 'aws-amplify';
 
-import LoaderButton from '../../components/LoaderButton';
+import {
+  selectNoteCreating,
+  selectNoteCreateError,
+  selectNoteCreated
+} from '../../store/selector/note';
+
+import {
+  createNoteAction,
+  clearNoteStateAction
+} from '../../store/reducers/note';
 
-import { s3Upload } from '../../libs/awsLib';
+import LoaderButton from '../../components/LoaderButton';
 
 import config from '../../config/aws.config';
 
 import { useNewNoteStyle } from './NewNote.style';
 
-const NewNote: FC<RouteComponentProps> = props => {
-  const file = useRef<File | null>(null);
+const NewNote: FC<RouteComponentProps> = ({ history }) => {
+  const { push: historyPush } = history;
```

Comment:
Why rename it tho, doesn't look like there is another push variable.

Reply:
More of a readability, `push` some might think its an `array prototype` or maybe a `lodash` or other library type of `push` just wanted to declare that its a `history push`